.github/workflows/build_and_test.yml

```diff
@@ -17,28 +20,44 @@ jobs:
         include:
           - image: ubuntu2204
             python: python3
+            configure: cmake -DPCAPPP_BUILD_EXAMPLES=ON -DPCAPPP_BUILD_TESTS=ON -DLIGHT_PCAPNG_ZSTD=OFF -S . -B $BUILD_DIR
+          - image: ubuntu2004
+            python: python3
+            configure: cmake -DPCAPPP_BUILD_EXAMPLES=ON -DPCAPPP_BUILD_TESTS=ON -DLIGHT_PCAPNG_ZSTD=OFF -S . -B $BUILD_DIR
           - image: ubuntu2004
             python: python3
+            configure: cmake -DPCAPPP_BUILD_EXAMPLES=ON -DPCAPPP_BUILD_TESTS=ON -DLIGHT_PCAPNG_ZSTD=OFF -S . -B $BUILD_DIR
+            use_zstd: false
           - image: ubuntu1804
             python: python3.8
+            configure: cmake -DPCAPPP_BUILD_EXAMPLES=ON -DPCAPPP_BUILD_TESTS=ON -DLIGHT_PCAPNG_ZSTD=OFF -S . -B $BUILD_DIR
           - image: centos7
             python: python3.8
             test-flags: --test-args="-s"
+            configure: cmake -DPCAPPP_BUILD_EXAMPLES=ON -DPCAPPP_BUILD_TESTS=ON -DLIGHT_PCAPNG_ZSTD=OFF -S . -B $BUILD_DIR
           - image: fedora34
             python: python3
+            configure: cmake -DPCAPPP_BUILD_EXAMPLES=ON -DPCAPPP_BUILD_TESTS=ON -DLIGHT_PCAPNG_ZSTD=OFF -S . -B $BUILD_DIR
           - image: alpine315
             python: python3
-            configure: --musl
+            configure: cmake -DPCAPPP_BUILD_EXAMPLES=ON -DPCAPPP_BUILD_TESTS=ON -DLIGHT_PCAPNG_ZSTD=OFF -S . -B $BUILD_DIR
 
     steps:
       - name: Checkout code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: "0"
+
+      - name: Mark repository as safe (Checkout actions is not run with same UID)
```

Comment:
I'm not sure I understand

Reply:
Introduced with new version of git in Ubuntu 22.04
https://github.com/actions/checkout/issues/760

The docker is executed with a different user ID as the actions/checkout so the repository safe trig an error